src/http/web_socket.cr

```diff
@@ -134,4 +141,57 @@ class HTTP::WebSocket
 
   def close
   end
+
+  # Opens a new websocket to the target host. This will also handle the handshake
+  # and will raise an exception if the handshake did not complete successfully.
+  #
+  # ```
+  # WebSocket.open("websocket.example.com", "/chat")              # Creates a new WebSocket to `websocket.example.com`
+  # WebSocket.open("websocket.example.com", "/chat", ssl = true)  # Creates a new WebSocket with SSL to `ẁebsocket.example.com`
+  # ```
+  def self.open(host, path, port = nil, ssl = false)
+    port = port || (ssl ? 443 : 80)
+    socket = TCPSocket.new(host, port)
+    socket = OpenSSL::SSL::Socket.new(socket) if ssl
+
+    headers = HTTP::Headers.new
+    headers["Host"] = "#{host}:#{port}"
+    headers["Connection"] = "Upgrade"
+    headers["Upgrade"] = "websocket"
+    headers["Sec-WebSocket-Version"] = VERSION.to_s
+    headers["Sec-WebSocket-Key"] = Base64.encode(StaticArray(UInt8, 16).new { rand(256).to_u8 })
+
+    path = "/" if path.empty?
+    handshake = HTTP::Request.new("GET", path, headers)
+    handshake.to_io(socket)
+    handshake_response = HTTP::Response.from_io(socket)
+    unless handshake_response.status_code == 101
+      raise "Handshake got denied. Status code was #{handshake_response.status_code}"
```

Comment:
Anyone got an idea if there's a more specific exception class we can reuse here? `SocketError` perhaps?